veer/src/main/java/frc/robot/control/ManualInputInterfaces.java

```diff
@@ -0,0 +1,165 @@
+// ************************************************************
+// Bishop Blanchet Robotics
+// Home of the Cybears
+// FRC - Rapid React - 2022
+// File: ManualInputInterfaces.java
+// Intent: Forms a class that grants access to driver controlled inputs.
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+package frc.robot.control;
+
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
+import frc.robot.commands.*;
+import frc.robot.*;
+import frc.robot.subsystems.*;
+
+public class ManualInputInterfaces
+{
+  // sets joystick variables to joysticks
+  private XboxController driverController = new XboxController(Constants.portDriverController); 
+  private XboxController coDriverController = new XboxController(Constants.portCoDriverController); 
+
+  // subsystems needed for inputs
+  private SubsystemCollection subsystemCollection = null;
+
+  /**
+   * The constructor to build this 'manual input' conduit
+   */
+  public ManualInputInterfaces(SubsystemCollection currentCollection)
+  {
+    subsystemCollection = currentCollection;
+  }
+
+  /**
+   * A method to access the driver controller
+   * @return XboxController for the driver.
+   */
+  public XboxController getDriverController(){
+    return driverController;
+  }
+
+  /**
+   * A method to get the arcade drive X componet being input from humans
+   * @return - a double value associated with the magnitude of the x componet
+   */
+  public double getInputArcadeDriveX()
+  {
+    return driverController.getLeftX();
+  }
+
+  /**
+   * A method to get the arcade drive X componet being input from humans
+   * @return - a double value associated with the magnitude of the x componet
+   */
+  public double getInputArcadeDriveY()
+  {
+    // need to invert the y for all xbox controllers due to xbox controler having up as negative y axis
+    return driverController.getLeftY() * -1.0;
+  }
+
+  /**
+   * A method to get the spin drive X componet being input from humans
+   * @return - a double value associated with the magnitude of the x componet
+   */
+  public double getInputSpinDriveX()
+  {
+    return driverController.getRightX();
+  }
+
+  /**
+   * A method to get the co-driver 'intake speed' input from humans
```

Comment:
I don't understand what this is for.  I thought the intake and layup speeds were constants.

Reply:
So in the original spec (I updated it a bit) I saw text that made me think that the Left Trigger and Right Trigger were being used for motor speed.  Since triggers on the XBox controller are variable (0.0 to 1.0), it seemed to make sense to me to have the motor speed be variable via this XBox control.  The need for instant on to a 'default speed' was a concern of mine too so I added the whole 'auto intake' and 'auto layup' buttons (A and Y).  As for the motor we are using, as best as I'm aware is cannot produce a constant velocity (unless we were to put an encoder on it) - so speed here is somewhat relative to voltage, current, wiring, etc.